### PR TITLE
Add Cut & Run preprocessing workflow

### PR DESCRIPTION
**Pull request by marcellevstek**
[REID-2129](https://genialis.atlassian.net/browse/REID-2129)

* [x] Update CHANGELOG.rst for each commit separately:
  * Pay attention to write entries under the "Unreleased" section.
  * Mark all breaking changes as "**BACKWARD INCOMPATIBLE:**" and put them
    before non-breaking changes.
  * If a commit modifies a feature listed under "Unreleased" section,
    it might be sufficient to modify the existing CHANGELOG entry from previous
    commit(s).
* [x] Bump the process version:
  * **MAJOR version (first number)**: Backward incompatible changes (changes
    that break the api/interface). Examples: renaming the input/output, adding
    mandatory input, removing input/output...
  * **MINOR version (middle number)**: add functionality or changes in a
    backwards-compatible manner. Examples: add output field, add non-mandatory
    input parameter, use a different tool that produces same results...
  * **PATCH version (last number)**: changes/bug fixes that do not affect
    the api/interface. Examples: typo fix, change/add warning messages...
* [x] All inputs are used in process.
* [x] All output fields have a value assigned to them.

[REID-2129]: https://genialis.atlassian.net/browse/REID-2129?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: resolwe_bio/processes/workflows/cut_and_run.py [hunk 1]
```diff
@@ -0,0 +1,478 @@
...
+                input={"alignment": deduplicate_alignment_spikein},
+            )
+
+            input_normalization = {
+                "alignment": deduplicate_alignment,
+                "bedpe": spikein_normfactor,
+                "scale": inputs.normalization_options.scale,
+            }
+
+            Data.create(
```
[the remark this answers is not in the document]
Comment (marcellevstek) [reply resolwe_bio/processes/workflows/cut_and_run.py:473]: The code in ``calculate-bigwig`` states that the scale factor is equal to 1 if no BEDPE input is given. This is what confused me.

File: resolwe_bio/processes/workflows/cut_and_run.py [hunk 2]
```diff
@@ -0,0 +1,478 @@
...
+                "stringency": inputs.trimming_options.adapter_options.stringency,
+                "error_rate": inputs.trimming_options.adapter_options.error_rate,
+            },
+        }
+
+        if inputs.trimming_options.nextseq:
+            input_trimgalore["quality_trim"][
+                "nextseq"
+            ] = inputs.trimming_options.nextseq
+        if inputs.trimming_options.adapter_options.adapter_1:
```
[the remark this answers is not in the document]
Comment (marcellevstek) [reply resolwe_bio/processes/workflows/cut_and_run.py:344]: Looks better :)

File: resolwe_bio/processes/workflows/cut_and_run_beta.py [hunk 3]
```diff
@@ -0,0 +1,460 @@
...
+                description="Magnitude of the scale factor. The scaling factor is calculated by "
+                "dividing the scale with the number of features in BEDPE (scale/(number of features)). "
+                "Default: 10000.",
+                disabled="normalization_options.skip_norm == True",
+            )
+
+        class SubsampleOptions:
+            """Subsampling options."""
+
+            n_reads = IntegerField(
```
Comment (jkokosar) [inline resolwe_bio/processes/workflows/cut_and_run_beta.py:277]: I think we should make downsampling optional - please add additional checkbox option for that. The default should be True.

File: resolwe_bio/processes/workflows/cut_and_run.py [hunk 4]
```diff
@@ -0,0 +1,478 @@
...
+                    "adapters and universal adapters.",
+                    required=False,
+                )
+                universal_adapter = StringField(
+                    label="Universal adapters",
+                    description="Instead of default detection use specific "
+                    "adapters. Use 13bp of the Illumina universal adapter, "
+                    "12bp of the Nextera adapter or 12bp of the Illumina "
+                    "Small RNA 3' Adapter. Selecting to trim smallRNA "
+                    "adapters will also lower the length value to 18bp. "
```
[the remark this answers is not in the document]
Comment (marcellevstek) [reply resolwe_bio/processes/workflows/cut_and_run.py]: I changed the label to reflect this: ``adapters will also lower the min length value to 18bp.``.  But I am not sure this is correct. The field description was obtained from the ``.yml`` workflow.

**Review by jkokosar — APPROVED** [review]

**Pull request merged**